enterprise/models.py

```diff
@@ -321,6 +368,147 @@ def __repr__(self):
         return self.__str__()
 
 
+class UserDataSharingConsentAudit(TimeStampedModel):
+    """
+    Object that exists to store the canonical state of whether a particular
+    user has given consent for their course data to be shared with a particular
+    single-sign-on identity provider
+    """
+    class Meta(object):
+        app_label = 'enterprise'
+        verbose_name = "Data Sharing Consent Audit State"
+        verbose_name_plural = "Data Sharing Consent Audit States"
+        unique_together = ('user', 'enterprise_customer',)
+
+    NOT_SET = 'not_set'
+    ENABLED = 'enabled'
+    DISABLED = 'disabled'
+    STATE_CHOICES = (
+        (NOT_SET, 'Not set'),
+        (ENABLED, 'Enabled'),
+        (DISABLED, 'Disabled'),
+    )
+
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+    )
+
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        on_delete=models.CASCADE,
+    )
```

Comment:
@haikuginger For now can we just throw an error if the EnterpriseCustomerUser does not yet exist (if the user is not yet linked to an EC)? I believe that at the moment, we are only supporting enterprise users that get manually linked to their EC by edX staff in the Django admin. I don't like an approach where we can model the data sharing consent for a user without also requiring that the user is linked to an EC. It invites data inconsistencies.